remix.init/setup-git-repository.js

```diff
@@ -1,30 +1,62 @@
-const { execSync } = require('child_process');
+const { execSync } = require("child_process");
 
-async function assertCommand(command, url) {
+function assertCommand(command, url) {
   try {
-    await execSync(`hash ${command}`, { stdio: 'pipe' })
+    execSync(`hash ${command}`, { stdio: "pipe" });
   } catch (error) {
-    console.log(`Please install '${command}' (${url}) and rerun the command.`)
+    console.log(`Please install '${command}' (${url}) and rerun the command.`);
 
     // Terminate setup process
-    process.exit(0)
+    process.exit(0);
   }
 }
 
-async function setupGitRepository(options) {
-  console.log(`Setting up repository: ${options.appName}... ⏰`)
+async function setupGitRepository(
+  appName,
+  environmentName,
+  azureLocation,
+  subscriptionId,
+  rootDirectory
+) {
+  console.log(`Setting up repository: ${appName}... ⏰`);
 
   // Check for necessary commands exists
-  await assertCommand('git', 'https://git-scm.com/book/en/v2/Getting-Started-Installing-Git')
-  await assertCommand('gh', 'https://cli.github.com/manual/installation')
+  assertCommand(
+    "git",
+    "https://git-scm.com/book/en/v2/Getting-Started-Installing-Git"
+  );
+  assertCommand("gh", "https://cli.github.com/manual/installation");
 
-  await execSync(`gh auth login --hostname github.com --git-protocol https --web`)
-  await execSync(`git init`)
-  await execSync(`git add .`)
-  await execSync(`git commit -m “Initial commit”`)
-  await execSync(`gh repo create ${options.appName} --public --push --source ./`)
+  console.log(`Auth with GitHub...`);
 
-  console.log(`Successfully setup repository 🎉`)
+  execSync(`gh auth login --hostname github.com --git-protocol https --web`, {
+    stdio: "inherit",
+    encoding: "utf8",
+  });
+  execSync(`git init`, {
+    cwd: rootDirectory,
+  });
+  execSync(`git add .`, {
+    cwd: rootDirectory,
+  });
+  execSync(`git commit -m "Initial commit"`, {
+    cwd: rootDirectory,
+  });
+
+  execSync(
+    `GH_DEBUG=1 gh repo create ${appName} --private --push --source ${rootDirectory}`,
+    {
+      cwd: rootDirectory,
+    }
+  );
+
+  execSync(`azd pipeline config`, {
```

Comment:
According to [Authorize GitHub to deploy to Azure](https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/configure-devops-pipeline?tabs=GitHub#authorize-github-to-deploy-to-azure): `azd pipeline config` also creates a private GitHub repository and pushes code to the new repo. Did we figure out what we need to to using `gh` command? Or should everything done using `azd pipeline config`?

Reply:
It won't if we're already in a Git repo.

Reply:
Removed the `gh` stuff completely. `azd pipeline config` and such takes care of that setup for us.